### ISSUE TITLE
 Chinese characters TAB width is incorrect

### ISSUE THREAD

**Issue by linlexing** [codemirror/CodeMirror#3996]
input:
```
a	b	c
中	b	c
```
Second row ‘b’ and ‘c’ not in the correct position

**Comment by marijnh** [comment]
What did you input? In your pasted content, there are no tabs, only spaces. Are you talking about the fact that the 'b c' on the second line is slightly more indented than those on the first line? If so, that's because the '中' is wider than an 'a' in your font, and CodeMirror tabs are represented as a (column-dependent) set of spaces.

**Comment by linlexing** [comment]
i modify the previous post,i see the **textarea** represented correct.
![-1](https://cloud.githubusercontent.com/assets/834303/15004029/75673308-11e4-11e6-99ee-7dbccbea6d4a.png)

**Comment by marijnh** [comment]
Yes, that's a limitation of the way CodeMirror draws spaces. Fixing this would require measuring the content before each tab in advance, which would be prohibitively slow (by causing lots of browser relayouts).

**Comment by linlexing** [comment]
The last part of an endeavor is the hardest to finish.

**Comment by themrmax** [comment]
If you use a monospaced Chinese font like SimSun, it seems like the chinese characters always take up 2 spaces and the roman characters always take up one space:

![image](https://cloud.githubusercontent.com/assets/8220288/20418951/6920d050-ada6-11e6-8b55-724b538dd2d4.png)

I wonder if there is a way in the code to check if a letter is a pictogram, and in that case reduce the number of spaces to indent accordingly?
